### PR TITLE
fix type annotation for pandas._config.config.py

### LINKED ISSUE #25899: Fix Type Annotations in pandas/_config/config.py
Part of #25882

Current errors are as follows:

```sh
pandas/_config/config.py:60: error: Need type annotation for '_deprecated_options'
pandas/_config/config.py:61: error: Need type annotation for '_registered_options'
pandas/_config/config.py:62: error: Need type annotation for '_global_config'
```

Should add annotations and remove this module from mypy.ini - think this is straightforward so community PR is certainly welcome!

### PR DESCRIPTION
**Pull request by vaibhavhrt**
also remove this module from mypy.ini

- [x] closes #25899 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by vaibhavhrt** [comment]
@WillAyd I am little confused about what type to use for namedtuple in `_deprecated_options`, `_registered_options`, please let me know if what I have done looks right.

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

File: pandas/_config/config.py [hunk 1]
```diff
@@ -51,15 +51,16 @@
...
 from contextlib import contextmanager
 import re
+from typing import Dict
 import warnings
 
 DeprecatedOption = namedtuple('DeprecatedOption', 'key msg rkey removal_ver')
 RegisteredOption = namedtuple('RegisteredOption',
                               'key defval doc validator cb')
 
-_deprecated_options = {}  # holds deprecated option metdata
```
Comment (jreback) [inline pandas/_config/config.py]: can you leave all of the comments, you can put them above each option

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_config/config.py (hunk 1 above)
Comment (vaibhavhrt) [reply pandas/_config/config.py]: no problem.

Comment (vaibhavhrt) [reply pandas/_config/config.py]: Alright, its done. Is it ok?

File: pandas/_config/config.py [hunk 2]
```diff
@@ -51,15 +51,19 @@
...
-_registered_options = {}  # holds registered option metdata
-_global_config = {}  # holds the current values for registered options
+# holds deprecated option metdata
+_deprecated_options = {}  # type: Dict[str, DeprecatedOption]
+# holds registered option metdata
+_registered_options = {}  # type: Dict[str, RegisteredOption]
+# holds the current values for registered options
+_global_config = {}  # type: Dict[str, str]
```
Comment (jreback) [inline pandas/_config/config.py]: can you add a blank line between the cases. also pls type _registered_keys

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_config/config.py (hunk 2 above)
Comment (vaibhavhrt) [reply pandas/_config/config.py]: `# holds registered option metdata` to `# holds registered_keys metdata` right?

Comment (jreback) [reply pandas/_config/config.py]: no there is another option called registered_keys below (so add a type for that one)

Comment (vaibhavhrt) [reply pandas/_config/config.py]: Ok, got it. I misunderstood what you meant by "type". I will get this done tomorrow too.

Comment (vaibhavhrt) [reply pandas/_config/config.py]: @jreback all done, take a look now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @vaibhavhrt

**Comment by vaibhavhrt** [comment]
@jreback My pleasure, looking forward to contribute more.

**Comment by WillAyd** [comment]
@vadakattu thanks! We would ideally be able to eliminate all items from being blacklisted in mypy.ini save maybe test folders for now, so if you see other modules you want to attack feel free to open an issue / PR for it

**Comment by vaibhavhrt** [comment]
@WillAyd looks like you mentioned someone else by mistake. But yes I would be happy to help. I think you have listed about 60+ modules in some issue. I will start working on them one by one. Is it okay if I create the issue for each one them too myself, or will you do that?
I am thinking if one single person can assign modules that would be good, because there can be multiple people working 'typing' and there can be situations when two people accidentally start working on same file. You don't have to create an issue, you can just leave a comment in the master issue that which module I should work on next, I can then take care of creating an issue and PR.

**Comment by WillAyd** [comment]
> Is it okay if I create the issue for each one them too myself, or will you do that?

Go ahead and create an issue when you have one in mind

**Comment by vaibhavhrt** [comment]
ok, I will do that.

**Pull request merged**